appserver/topology/graph.go

```diff
@@ -0,0 +1,124 @@
+package topology
+
+import (
+	"encoding/json"
+)
+
+// Graph contains the groupds, edges and nodes of the graph.
+type Graph struct {
+	Nodes  []Node  `json:"nodes,omitempty" protobuf:"bytes,1,opt,name=nodes"`
+	Edges  []Edge  `json:"edges,name=edges"`
+	Groups []Group `json:"groups,name=groups"`
+}
+
+// Node is the id and name of a node.
+type Node struct {
+	ID   string `json:"id,name=id"`
+	Name string `json:"name,name=name"`
+}
+
+// Edge is the source and target of a node.
+type Edge struct {
+	Source string `json:"source,name=source"`
+	Target string `json:"target,name=target"`
```

Comment:
`Edge` should also have `ID` and `TYPE`  keys.
eg: 
id: "8f6dacd7-8e6e-11e9-bd8e-0a580a80002b_4edcad80-8e52-11e9-bd8e-0a580a80002b"
source: "8f6dacd7-8e6e-11e9-bd8e-0a580a80002b"
target: "4edcad80-8e52-11e9-bd8e-0a580a80002b"
type: "connects-to"

Reply:
Will add this! Out of curiosity, would we really need this value here? Wouldn't the type always be "connects-to"..?

Reply:
Based on this type `connects-to` we are currently showing the unidirectional arrow from source to target. later on, if we need a bidirectional arrow or some other link to show a different edge visualization (eg: network link between the nodes) this `type` attribute would help us.

Reply:
Thanks Karthik!

Reply:
Thanks Karthik!

Reply:
Thanks Karthik! fixed here: 6ff654d7cdc9847de580d2d73acb872489dde13f